File: priv/sample.js

var pattern = window.matches.pattern;
var start2 = pattern({
	'x,y': function(x,y) {
		var f = function(a) {
		return a.log('100');
	};
		f(console);
		return (pattern({
	'1': function() {
		return console.log([x,y]);
	},
	'_': function() {
		return console.log('ok');
	}})
)(x);
	}});
var start = pattern({
	'': function() {
		start2(1,3);
		var j = 5;
		var n = fac(j);
		return console.log('factorial ~p',[j,n]);
	}});
var fac = pattern({
	'0': function(_0) {
		return 1;
	},
	'n': function(n) {
		return n * fac(n - 1);
	}});
start();

File: src/sample.erl
-module(sample).
-compile({parse_transform, shen}).
-compile(export_all).

-output("priv").
-js([start/0,start2/2,fac/1]).
-jsmacro([macro/3]).

macro(A,B,C) ->
    X = document,
    jq(X),
    jq("document"),
    ws:send('Bert':encodebuf(
        [{source,'Bert':binary(A)},
         {x,C},
         {pickle,'Bert':binary(B)},
         {linked,C}])).

start2(X,Y) -> 
    case X of
        1 -> console:log([X,Y]);
        _ -> console:log("ok") end.

start() ->
    start2(1,3),
    J = 5,
    N = fac(J),
    console:log("factorial ~p", [J, N]).

fac(0) -> 1;
fac(N) -> N * fac(N-1).

main() ->
    A = "1",
    B = "2",
    Script = macro(A,B,"3"),
    io:format("JS Macro: ~s",[Script]).

